src/Roslyn.Diagnostics.Analyzers/Core/CreateTestAccessorFixer.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Roslyn.Diagnostics.Analyzers
+{
+    public abstract class CreateTestAccessorFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(RoslynDiagnosticIds.CreateTestAccessorRuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            // This is a refactoring for one-off test accessor creation. Batch fixing is disabled.
```

Comment:
If we do not support FixAll and also the diagnostic is a non-configurable hidden diagnostic, what is the benefit of writing this as an analyzer/fixer instead of just writing a CodeRefactoringProvider?

Reply:
CodeRefactoringProvider does not exist

Reply:
I meant http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/CodeRefactorings/CodeRefactoringProvider.cs,11

Reply:
➡️ This is not currently an option, but hopefully someday https://github.com/dotnet/roslyn/pull/35685 makes it viable.

Reply:
Is this a valid possibility now?

Reply:
Yes, this could be converted in the future.